polynote-frontend/polynote/ui/component/notebook/cell.ts

```diff
@@ -878,6 +879,8 @@ export class CodeCell extends Cell {
         updateError(this.state.error)
         cellState.observeKey("error", error => updateError(error));
 
+        let nbStatusObs: IDisposable | undefined = undefined;
```

Comment:
Reading this "`let` that async things mutate" I get that it's kind of hard for the person who didn't write it to reason about it. I wonder if we can find something that's more reasonable but isn't react.

Reply:
(to be clear, I have done this too in several places, but those are familiar places which are hard to look at objectively)

Reply:
Agreed, would love to find an alternative to this pattern for sure